aiven/resource_service.go

```diff
@@ -574,6 +575,16 @@ func resourceServiceDelete(d *schema.ResourceData, m interface{}) error {
 	client := m.(*aiven.Client)
 
 	projectName, serviceName := splitResourceID2(d.Id())
+
+	service, err := client.Services.Get(projectName, serviceName)
+	if err != nil {
+		return err
+	}
+
+	if service.TerminationProtection == true {
```

Comment:
What's the reason for doing this check on client side? I don't know if there's any harm in doing that really but since the server should anyway reject the deletion this seems a bit unnecessary and was just wondering why it's been added.

Reply:
It was requested here https://github.com/aiven/terraform-provider-aiven/issues/154 to have such protection, and from the name of the field itself, it wasn't clear that it will prevent from deletion of a service. After doing writing acceptance tests for this feature, I understood that `termination_protection` does already what the customer wants, and there are not that many good reasons to have this in TF provider code.

The few I can think of are:
- test during the run of `make testacc` weather `termination_protection` works on API side and block deletion. 
- if by any reason API `termination_protection` deletion protection fails we have a second line of defence in TF provider.

I still decided to make this PR since code was ready, knowing that it might not get merged in a current form of at all. Therefore if you do not see any benefits of having this in TF provider feel free to decline this PR. Or maybe we can just leave ACC tests to validation this functionality working on API side. 

Let me know WDYT?

Reply:
I think having the test makes sense but I'd drop this check from the provider and just have the test ensure that the server actually does prevent deleting the service.

Reply:
only ACC tests left which validate API deletion protection with `termination_protection` set true